sycl/doc/extensions/LevelZeroBackend/LevelZeroBackend.md

```diff
@@ -0,0 +1,126 @@
+|``` make<context>(const vector_class<device> &, ze_context_handle_t);```| Constructs a SYCL context instance from a Level-Zero ```ze_context_handle_t```. The context is created against the devices passed in. There must be at least one device given and all the devices must be from the same SYCL platform and thus from the same Level-Zero driver.|
+|``` make<queue>(const context &, ze_command_queue_handle_t);```| Constructs a SYCL queue instance from a Level-Zero ```ze_command_queue_handle_t```. The context argument must be a valid SYCL context encapsulating a Level-Zero context. The queue is attached to the first device in the passed SYCL context.|
+|``` make<program>(const context &, ze_module_handle_t);```| Constructs a SYCL program instance from a Level-Zero ```ze_module_handle_t```. The context argument must be a valid SYCL context encapsulating a Level-Zero context. The Level-Zero module must be fully linked (i.e. not require further linking through [```zeModuleDynamicLink```](https://spec.oneapi.com/level-zero/latest/core/api.html?highlight=zemoduledynamiclink#_CPPv419zeModuleDynamicLink8uint32_tP18ze_module_handle_tP28ze_module_build_log_handle_t)), and thus the SYCL program is created in the "linked" state.|
+
+NOTE: We shall consider adding other interoperability as needed, if possible.
+                
+### 4.4 Level-Zero handles' ownership and thread-safety
+        
+The Level-Zero runtime doesn't do reference-counting of its objects, so it is crucial to adhere to these
+practices of how Level-Zero handles are manged.
+                
+#### 4.4.1 SYCL runtime takes ownership
+                
+Whenever the application creates a SYCL object from the corresponding Level-Zero handle via one of the ```make<T>()``` functions,
+the SYCL runtime takes ownership of the Level-Zero handle. The application must not use the Level-Zero handle after
+the last host copy of the SYCL object is destroyed (as described in the core SYCL specification under
+"Common reference semantics"), and the application must not destroy the Level-Zero handle itself.
+                                                                
+#### 4.4.2 SYCL runtime assumes ownership
+
+The application may call the ```get_native<T>()``` member function of a SYCL object to retrieve the underlying Level-Zero handle,
+however, the SYCL runtime continues to retain ownership of this handle. The application must not use this handle after
+the last host copy of the SYCL object is destroyed (as described in the core SYCL specification under
+"Common reference semantics"), and the application must not destroy the Level-Zero handle.
+
+#### 4.4.3 Considerations for multi-threaded environment
+
+The Level-Zero API is not thread-safe, but it is free-threaded meaning that only the access to its global/shared state is guarded for thread-safety.

```

Comment:
I find this sentence to be confusing.  Is the term "free-threaded" well defined and something that most people understand?  I've been doing MT programming for a long time, and I don't think I've seen it before today.

The description is confusing because it says that access to the Level Zero global/shared state is internally locked.  If that was really true, then wouldn't the driver be thread-safe?  The documentation says it is not thread-safe, though, so I think this statement must not be true.  Or, maybe there is some confusing definition of what "global / shared state" means.

If the point we are trying to make is that Level Zero is not thread-safe, why not just say that directly:

> The Level-Zero API is not thread-safe.

Is there some additional information we are trying to convey with the part about free-threading?

Reply:
I wanted to convey more that just "not thread-safe" (will add a link to Level-Zero spec). The incoming arguments are what need addition locks, and shared state is locked by the driver. It is essentially making it possible, with the proper locking in the app, to use Level-Zero in a multi-threaded environment.

Reply:
> It is essentially making it possible, with the proper locking in the app, to use Level-Zero in a multi-threaded environment.

By this statement, do you mean it is possible to use Level-Zero in a SYCL application that is multi-threaded, or are you just saying that it is possible to write a multi-threaded app that uses Level-Zero (and does not use SYCL)?

If it is just that later statement that you mean, how is this relevant to the SYCL interop API?

Reply:
I think it is possible to use Level-Zero from a SYCL app.